src/CreatePaletteForm/CreatePaletteForm.js

```diff
@@ -35,24 +35,33 @@ export class CreatePaletteForm extends Component {
       color4: currentPalette[3].hexCode,
       color5: currentPalette[4].hexCode
     }
-    await postNewPalette(postPalette);
-    const updatePalettes = await getSelectedPalettes(projectId);
-    setSelectedPalettes(updatePalettes)
+    try {
+      await postNewPalette(postPalette);
+      const updatePalettes = await getSelectedPalettes(projectId);
+      setSelectedPalettes(updatePalettes);
+    } catch ({message}) {
+      hasErrored(message)
+    }
+    
     this.setState({ name: "" })
   }
 
   handleSaveEdits = async () => {
-    const { currentPalette, currentPaletteId, setSelectedPalettes, clearSelectedPaletteId, setRandomPalette } = this.props;
+    const { currentPalette, currentPaletteId, setSelectedPalettes, clearSelectedPaletteId, setRandomPalette, hasErrored } = this.props;
     const paletteToEdit = {
       color1: currentPalette[0].hexCode,
       color2: currentPalette[1].hexCode,
       color3: currentPalette[2].hexCode,
       color4: currentPalette[3].hexCode,
       color5: currentPalette[4].hexCode
     }
-    await editPalette(paletteToEdit, currentPaletteId.id)
-    const updatePalettes = await getSelectedPalettes(currentPaletteId.projectId);
-    setSelectedPalettes(updatePalettes);
+    try {
```

Comment:
We'll probably have to add some additional tests to check both happy and sad paths. Looks great!